src/main/java/org/jvnet/hudson/plugins/repositoryconnector/ArtifactResolver.java

```diff
@@ -124,7 +143,22 @@ private boolean download(AbstractBuild<?, ?> build, BuildListener listener, fina
                 final String targetFileName = TokenMacro.expandAll(build, listener, a.getTargetFileName());
 
                 String version = TokenMacro.expandAll(build, listener, a.getVersion());
+
                 version = checkVersionOverride(build, listener, groupId, artifactId, version);
+                
+                if(version==null) {
+                	version = "LATEST";
+                	/**
+                	List<Version> versions = aether.resolveVersions(groupId, artifactId);
+                	for(Version v : versions) {
+                		logger.println("available version: "+v.getClass().getName()+" / "+v.toString());
+                	}
+                	if(versions.size()>0) {
+                		version = versions.get(versions.size()-1).toString();
+                		logger.println("no version set, resolved version: "+version);
+                	}
+                	**/
```

Comment:
the thing here is - the code there would actually DO what "LATEST" promises ... i.e. check what the latest version actually is ... so this would actually do "LATEST" more correctly